### PR TITLE
added take first and last

### PR DESCRIPTION
**Pull request by VinceDio**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -37,5 +37,22 @@ using StringThings.Extensions;
 "ABC".ContainsIgnoreCase("abc");
 ```
 
+
+## Take First/Last Characters Extensions
+No more length errors when using substring to get the leading or trailing characters from a string. Let this extension do the work for you of verifying that your substring doesn't exceed the length of the string you are evaluating.
```
Comment (dlumpp) [inline README.md:42]: I really appreciate you matching my energy with this text 😂

File: README.md [hunk 2]
```diff
@@ -37,5 +37,22 @@ using StringThings.Extensions;
...
+
+Prevents this error:
+`Index and length must refer to a location within the string.`
+
+```cs
+using StringThings.Extensions;
+// all return true
```
Comment (dlumpp) [inline README.md]: this comment isn't right here anymore

File: README.md [hunk 3]
```diff
@@ -37,5 +37,22 @@ using StringThings.Extensions;
...
+"principal".TakeLast(3) *italic*returns "pal"*italic*
+"ABC-123".TakeFirst(3) *italic*returns "ABC"*italic*
+"tiny".TakeLast(5) *italic*returns "tiny" (with no error!)*italic*
+"This is a sentence".TakeFirst(8) *italic*returns "This is "*italic*
```
Comment (dlumpp) [inline README.md]: looks like the italic markdown doesn't work within a code block. you can switch branches on the main page and see how the readme renders. I think we can just make these line comments

File: StringThings/Extensions/IgnoreCaseStringExtensions.cs [hunk 4]
```diff
@@ -12,7 +12,7 @@ private static void ThrowIfArgumentNull(string argName, string? s)
 
         public static bool EqualsIgnoreCase(this string? s, string? other)
         {
-            ThrowIfArgumentNull(nameof(s), s);
+            ThrowIfArgumentNull(nameof(s), s);
```
Comment (dlumpp) [inline StringThings/Extensions/IgnoreCaseStringExtensions.cs]: I took out this whitespace and the tests still pass 😉

File: StringThings.UnitTests/Extensions/TakeFirstStringExtensionsTestFixture.cs [hunk 5]
```diff
@@ -0,0 +1,44 @@
+﻿using FluentAssertions;
+using System;
+using Xunit;
+using StringThings.Extensions;
+
+namespace StringThings.UnitTests.Extensions
+{
+    public class TakeFirstStringExtensionsTestFixture
```
Comment (dlumpp) [inline StringThings.UnitTests/Extensions/TakeFirstStringExtensionsTestFixture.cs:8]: beautiful tests! did they feel useful to you??

**Review by dlumpp — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (VinceDio) [reply README.md:42]: "And YOU get a string, and YOU get a string!"

File: StringThings/Extensions/IgnoreCaseStringExtensions.cs (hunk 4 above)
Comment (VinceDio) [reply StringThings/Extensions/IgnoreCaseStringExtensions.cs]: Just practicing social distancing for COVID.

File: StringThings.UnitTests/Extensions/TakeFirstStringExtensionsTestFixture.cs (hunk 5 above)
Comment (VinceDio) [reply StringThings.UnitTests/Extensions/TakeFirstStringExtensionsTestFixture.cs:8]: They have their purpose. For this situation where you are making a reusable public library, good testing is essential and this type of extension lends itself to simple unit tests.

File: README.md (hunk 2 above)
Comment (VinceDio) [reply README.md]: Removed

File: README.md (hunk 3 above)
Comment (VinceDio) [reply README.md]: Replaced with comments

**Review by dlumpp — APPROVED** [review]

**Pull request merged**